mozilla/blurts-server
# Implement RP metrics for landing page CTA experiment (MNTOR-3341)

# References:

Jira: [MNTOR-3341](https://mozilla-hub.atlassian.net/browse/MNTOR-3341)



# Description

Follow-up for PR #4787 that makes sure we are passing the correct parameters to the Mozilla accounts endpoint `/authorization` and is implemented as [suggested by data science](https://mozilla.slack.com/archives/C05TV26C5V4/p1721156802024629?thread_ts=1719862765.251169&cid=C05TV26C5V4).

**1. Fetch `/metrics-flow` endpoint**
The parameters documented [here](https://mozilla.github.io/ecosystem-platform/relying-parties/reference/metrics-for-relying-parties#relying-party-hosted-email-form), plus `service` set to Monitor's OAuth client ID.

**2. Mozilla accounts sign-in/up URL**
- `flow_id`, `flow_begin_time`, and `device_id` are being set dynamically based on the `flowId`, `flowBeginTime`, and `deviceId` values returned from the `/metrics-flow` call in step `1`.
- `entrypoint` set consistently
- `entrypoint_experiment` and `entrypoint_variation` set based on the active experiment (if any).
- UTM parameters set as appropriate (same as described above for the metrics-flow call).

# How to test

1. Enable the Nimbus experiment `landing-page-free-scan-cta`
4. Signing in/up with one of the “Free scan” CTA on the landing page.
5. The relevant URL parameter should be present on the Mozilla accounts auth page.


[MNTOR-3341]: https://mozilla-hub.atlassian.net/browse/MNTOR-3341?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
